### ISSUE TITLE
Add Kaggle Playground Series S6E4 example notebook

### ISSUE THREAD

**Issue by sunkusun9** [sunkusun9/ml-labs#109]
Create an example Jupyter notebook for Kaggle Playground Series Season 6 Episode 4.

This notebook will serve as a practical usage guide and a vehicle for discovering areas where the package can be improved after the v0.7.0 multiprocessing changes.

**Comment by sunkusun9** [comment]
Fix all collectors: suppress errors when node data is absent (finalized before collect)

**ModelAttrCollector**: `_load_results` returns `None` on missing file; `get_attr` / `get_attrs_agg` propagate `None`

**MetricCollector**: same `_load_results` fix; `get_metric` returns `None`; `get_metrics` filters out `None` results and returns `None` if nothing remains

**SHAPCollector**: `get_feature_importance_agg` returns `None` when no outer folds collected

**OutputCollector**: `get_output` returns `None` on missing file; `get_outputs` returns `{}` instead of raising; removed stray debug `print`

**ProcessCollector**: `get_output` returns `None` when no node data available

**Comment by sunkusun9** [comment]
Add `Experimenter.get_collect_status(collector, nodes=None)`

Returns `{node_name: status}` for all head nodes matched by the collector's connector.

Statuses: `'collected'` / `'not_collected'` / `'finalized'` (finalized before collect ran) / `'error'`

**Comment by sunkusun9** [comment]
## Changes from S6E4 notebook work

### sample_weight support
Added `sample_weight` as a first-class edge key. When `'sample_weight'` is present in node edges, it is automatically passed to `fit()` via the adapter pipeline.

- `adapter/_base.py`: `get_fit_params` unwraps and passes `sample_weight` from `train_data`
- `adapter/_lightgbm.py`: polars→pandas conversion for `sample_weight`
- `adapter/_catboost.py`: polars→pandas conversion for `sample_weight` (old CatBoost compat)
- `nn/_input.py`, `nn/_estimator.py`: `sample_weight` propagated through `_make_tf_dataset`, `fit`, `_fit`, `_split_val`

Usage:
```python
pipeline.set_node('xgb_node', edges={
    'X': [(None, features)],
    'y': [(None, 'target')],
    'sample_weight': [(None, 'weight_col')],
})
```

### Bug fix: LMAdapter multiclass coef concatenation
`adapter/_sklearn.py` `LMAdapter._get_coef`: `np.expand_dims(intercept_, axis=0)` → `axis=1`.
With multiclass LogisticRegression, `coef_` shape is `(n_classes, n_features)` and `intercept_` is `(n_classes,)`, so expand along axis=1 to get `(n_classes, 1)` before concatenating.

### Docs: GPU and multiprocessing
- `docs/guide/pipeline-experimenter.md`: Added "Parallel Execution and GPU" section — worker assignment table, node routing rules, `_needs_gpu` behavior, mixed GPU/CPU example
- `docs/guide/adapters.md`: Added "GPU Settings" section — `gpu` param values, per-adapter injection table, per-framework examples

Key clarification documented: nodes require `'gpu': 'yes'` in params to be routed to a GPU worker in multi-job mode. Without it, `get_gpu_usage` returns `GPU_NO` and the node runs on a CPU worker regardless of `gpu_id_list`.
